samples/RoutingSample.Web/UseEndpointRoutingStartup.cs

```diff
@@ -26,65 +22,59 @@ public void ConfigureServices(IServiceCollection services)
-                    new MatcherEndpoint((next) => (httpContext) =>
-                        {
-                            var response = httpContext.Response;
-                            var payloadLength = _helloWorldPayload.Length;
-                            response.StatusCode = 200;
-                            response.ContentType = "text/plain";
-                            response.ContentLength = payloadLength;
-                            return response.Body.WriteAsync(_helloWorldPayload, 0, payloadLength);
-                        },
-                         RoutePatternFactory.Parse("/plaintext"),
-                        0,
-                        EndpointMetadataCollection.Empty,
-                        "Plaintext"),
-                    new MatcherEndpoint((next) => (httpContext) =>
-                        {
-                            var response = httpContext.Response;
-                            response.StatusCode = 200;
-                            response.ContentType = "text/plain";
-                            return response.WriteAsync("WithConstraints");
-                        },
-                        RoutePatternFactory.Parse("/withconstraints/{id:endsWith(_001)}"),
-                        0,
-                        EndpointMetadataCollection.Empty,
-                        "withconstraints"),
-                    new MatcherEndpoint((next) => (httpContext) =>
-                        {
-                            var response = httpContext.Response;
-                            response.StatusCode = 200;
-                            response.ContentType = "text/plain";
-                            return response.WriteAsync("withoptionalconstraints");
-                        },
-                        RoutePatternFactory.Parse("/withoptionalconstraints/{id:endsWith(_001)?}"),
-                        0,
-                        EndpointMetadataCollection.Empty,
-                        "withoptionalconstraints"),
-                });
-
-            services.TryAddEnumerable(ServiceDescriptor.Singleton<EndpointDataSource>(endpointDataSource));
         }
 
         public void Configure(Microsoft.AspNetCore.Builder.IApplicationBuilder app)
         {
-            app.UseEndpointRouting();
+            app.UseEndpointRouting(builder =>
+            {
+                builder.MapHello("/helloworld", "World");
+
+                builder.MapEndpoint(
+                    (next) => (httpContext) =>
+                    {
+                        var response = httpContext.Response;
+                        var payloadLength = _homePayload.Length;
+                        response.StatusCode = 200;
+                        response.ContentType = "text/plain";
+                        response.ContentLength = payloadLength;
+                        return response.Body.WriteAsync(_homePayload, 0, payloadLength);
+                    },
+                    "Home",
+                    "/");
```

Comment:
My thinking is MapEndpoint is for mapping a raw invoker.

MVC could add overloads for MapMvcEndpoint which would have MVC specific params. Need to think about how to get it to perform the logic that MvcDataSourceEndpoint runs for resolving MVC params to MVC actions and then create endpoints.

Reply:
What's the signature here, basically the same as `Use(...)` - seems pretty right for a first go.